### ISSUE TITLE
False Positive: Trivial `super.equals` delegation and benchmark-specific `Comparable` exceptions are treated as real `equals`/`hashCode` contract violations.

### ISSUE THREAD

**Issue by Carlson-JLQ** [checkstyle/checkstyle#19386]
I have read check documentation: https://checkstyle.org/checks/xxxxxx/nameofaffectedcheck.html
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```bash
/var/tmp $ javac YOUR_FILE.java
#[[MAKE SURE THERE IS SUCCESSFUL COMPILATION]]

/var/tmp $ cat config.xml
#[[PLACE YOUR OUTPUT HERE]]

/var/tmp $ cat YOUR_FILE.java
#[[PLACE YOU OUTPUT HERE]]

/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java $RUN_LOCALE -jar checkstyle-X.XX-all.jar -c config.xml YOUR_FILE.java
#[[PLACE YOUR OUTPUT HERE]]
```

For Windows users, please use `type` instead of `cat` and run# False Positive: Trivial `super.equals` delegation and benchmark-specific `Comparable` exceptions are treated as real `equals`/`hashCode` contract violations.

Version
unknown in repository metadata

## Checker
- Checker id: `EqualsHashCodeCheck`
- Checker description: Detects when a class defines an equals method but not a hashCode method, or vice versa.

## Description of the false positive
`diff/output/checkstyle.json` lists three false positives for `EqualsHashCodeCheck`, and all three come from one limitation in the checker's current matching logic: it is entirely syntactic. The implementation only asks whether a class contains a concrete `equals(Object)` declaration and lacks a concrete zero-argument `hashCode()` declaration.

That misses an important semantic distinction. Some `equals(Object)` methods do not establish a new equality contract at all, and some benchmark samples rely on extra policy assumptions that the checker does not model.

### `NegCase3.java`
This override is pure delegation to `Object.equals`. It does not compare fields, does not redefine equality, and therefore does not create a new requirement for a matching class-local `hashCode()` implementation.

```java
// A Java class that declares an equals method with trivial delegation to Object.equals should not be flagged as overriding equals without hashCode.
package scensct.core.neg;

public class NegCase3 {
    private String data;
    
    public NegCase3(String d) {
        this.data = d;
    }
    
    // Trivial equals that just delegates to super.equals
    @Override
    public boolean equals(Object obj) {
        return super.equals(obj);  // Single statement delegating to Object.equals
    }
}
```

This is the strongest false-positive example in the set. The checker sees a concrete `equals(Object)` method and stops there, even though the method body preserves identity equality exactly.

### `NegCase2.java`
Under the benchmark's intended semantics, this case should be ignored because the `equals(Object)` method is being treated as a bridge-like artifact related to `Comparable`.

```java
// A non-Comparable class contains a bridge method named "equals" with a single parameter of type Object should not be flagged as overriding equals without hashCode.
package scensct.core.neg;

public class NegCase2 implements Comparable<NegCase2> {
    // Bridge method generated by compiler for generic scenarios.
    // This is a synthetic/bridge method; the checker should ignore it.
    public boolean equals(Object obj) { // [REPORTED LINE]
        return false;
    }
    // The presence of this bridge method alone should not trigger the checker.
    public int compareTo(NegCase2 other) {
        return 0;
    }
}
```

In source form, though, this is an ordinary concrete `equals(Object)` declaration, not a real synthetic bridge method. `EqualsHashCodeCheck` cannot distinguish “compiler-generated bridge” intent from an actual source-level override because it only inspects the parsed source tree. So this sample is better understood as a benchmark-policy mismatch than as a strong checker bug.

### `NegCase7.java`
This sample assumes that implementing `Comparable` should suppress the rule even when the class also defines `equals(Object)` without `hashCode()`.

```java
// A class that implements Comparable should not be flagged as overriding equals without hashCode.
package scensct.core.neg;

public class NegCase7 implements Comparable<NegCase7> {
    // Overrides equals without hashCode, but class implements Comparable.
    // Checker should allow this.
    @Override
    public boolean equals(Object obj) {
        return false;
    }
    
    @Override
    public int compareTo(NegCase7 other) {
        return 0;
    }
}
```

That carveout is not part of the rule description or the Java `equals`/`hashCode` contract. A class can implement `Comparable` and still be inconsistent if it overrides `equals` without overriding `hashCode`. For that reason, this case also looks more like a benchmark-specific expectation than a robust false positive in the checker itself.

## Cause analysis
The real issue exposed by these samples is that `EqualsHashCodeCheck` reasons only about method signatures and declaration presence:

- if a class has a concrete `equals(Object)`, it expects a concrete `hashCode()`;
- if a class has a concrete `hashCode()`, it expects a concrete `equals(Object)`.

That works for the canonical contract violation, but it is too coarse for semantic edge cases such as pure delegation to `super.equals(obj)`. In `NegCase3.java`, the checker should ideally recognize that the override does not define a new equality relation and therefore should not trigger the rule.

The other two cases do not expose the same kind of semantic bug as clearly. They depend on assumptions that are external to the checker's documented behavior:

- treating a handwritten source method as if it were a compiler-synthesized bridge method;
- treating `Comparable` as an automatic exemption from the `equals`/`hashCode` pairing rule.

So the actionable checker gap is narrow but real: it has no exemption for behavior-preserving delegation overrides such as `return super.equals(obj);`.

## References
None known.

```cmd
set RUN_LOCALE="-Duser.language=en -Duser.country=US"
java %RUN_LOCALE% -jar checkstyle-X.XX-all.jar -c config.xml YOUR_FILE.java
```

in place of the last 2 commands above.

---

**Describe what you expect in detail.**

---

Still not clear ???
see example - https://checkstyle.org/report_issue.html#How_to_report_a_bug.3F

ATTENTION: FAILURE TO FOLLOW THE ABOVE TEMPLATE WILL RESULT IN THE ISSUE BEING CLOSED.

**Comment by romani** [comment]
please follow isssue template
